core/base/identifiers/Identifiers.h

```diff
@@ -887,6 +892,12 @@ namespace ttk {
       int height
         = static_cast<int>((globalBounds[3] - globalBounds[2]) / spacing_[1])
           + 1;
+      int depth
+        = static_cast<int>((globalBounds[5] - globalBounds[4]) / spacing_[2])
+          + 1;
+
+      // store global grid dimensions
+      this->globalDims_ = {width, height, depth};
```

Comment:
This line is not necessarily executed, in the case the global ids were created previously, so this line may never be reached.

I also just noticed that the global boundary is computed twice: in `GenerateGlobalIds` and in `preconditionGlobalBoundary`, that computes the global boundary in case boundary preconditioning needs to be done. From what I understood, with your PR, the computation of the global boundary should always be done, to create the meta grid.

If it is moved before GenerateGlobalIds, the computation is not done twice. The computation of `globalDims_` could be done at the same time.
What do you think?

Reply:
If I understand correctly, you suggest to move the computation of the global dimensions out of `Identifiers`, which may not be called if global identifiers are present on cells & vertices?

Reply:
I copied the `Identifiers` code inside `ImplicitTriangulation` and now `ImplicitTriangulation::createMetaGrid` only depends on the distributed dataset bounds.